jbake-core/src/main/java/org/jbake/parser/AsciidoctorEngine.java

```diff
@@ -98,7 +98,11 @@ public void processHeader(final ParserContext context) {
 
             if (hasJBakePrefix(key)) {
                 String pKey = key.substring(6);
-                documentModel.put(pKey, value);
+                if(value instanceof String) {
+                    processHeader(pKey, (String) value, documentModel);
+                } else {
+                    documentModel.put(pKey, value);
```

Comment:
We could always call `processHeader`. As the last case in that method adds the value for the given key to the documentModel Map. See line 284 in `MarkupEngine`.

Reply:
I decided to special case the "String" case, as I don't want to interfere with the asciidoc engine. As `NULL` could be generated, we'd still need a guard as in 
```java
if(value != null) {
    processHeader(pKey, value.toString(), documentModel);
}
```
I have zero experience with asciidocj, so can't predict if other types, than string are possible.

Reply:
Yes you are right. asciidoctorj takes a Map<Sting, Object> for attributes. So it would be worth a try. On the other hand we could postpone it for a later cleanup session.